### DISCUSSION TITLE
run-wasm-example.sh not working | 'remaining data [25]'

### DISCUSSION THREAD

**Discussion by sotrh** [gfx-rs/wgpu#2115] [Q&A] [answered]
I'm working on getting wgpu running on the web for my tutorial, but I'm not able to get wasm-bindgen working. I tested hello-triangle as well as cube with the same results.

wasm-bindgen --version: 0.2.76
rustc version: 1.55.0 (c8dfcfe04 2021-09-06)
cargo version: 1.55.0 (32da73ab1 2021-08-23)
os: Linux Mint

I didn't post this as an issue because I'm pretty sure it's a configuration issue on my end.

**Comment by kvark** [comment]
Could you post the exact log or output?

**Reply by sotrh** [reply]
RUST_BACKTRACE=1 ./run-wasm-example.sh cube
Compiling...
    Finished dev [unoptimized + debuginfo] target(s) in 0.16s
Generating bindings...
thread 'main' panicked at 'remaining data [25]', /home/benjamin/.cargo/registry/src/github.com-1ecc6299db9ec823/wasm-bindgen-cli-support-0.2.76/src/descriptor.rs:110:9
stack backtrace:
   0: rust_begin_unwind
             at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/panicking.rs:515:5
   1: std::panicking::begin_panic_fmt
             at /rustc/c8dfcfe046a7680554bf4eb612bad840e7631c4b/library/std/src/panicking.rs:457:5
   2: wasm_bindgen_cli_support::descriptors::execute
   3: wasm_bindgen_cli_support::Bindgen::generate_output
   4: wasm_bindgen_cli_support::Bindgen::generate
   5: wasm_bindgen::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

**Reply by sotrh** [reply]
My only guess is that `Cargo.lock` seems to be pointing to the wrong version of "wasm-bindgen".
```toml
[[package]]
name = "wasm-bindgen"
version = "0.2.78"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "632f73e236b219150ea279196e54e610f5dbafa5d61786303d4da54f84e47fce"
dependencies = [
 "cfg-if 1.0.0",
 "wasm-bindgen-macro",
]

[[package]]
name = "wasm-bindgen-backend"
version = "0.2.78"
source = "registry+https://github.com/rust-lang/crates.io-index"
checksum = "a317bf8f9fba2476b4b2c85ef4c4af8ff39c3c7f0cdfeed4f82c34a880aa837b"
dependencies = [
 "bumpalo",
 "lazy_static",
 "log",
 "proc-macro2",
 "quote",
 "syn",
 "wasm-bindgen-shared",
]
```

**Reply by sotrh** [reply]
In wgpu's Cargo.toml it's supposed to be using "0.2.76".
```toml
[target.'cfg(target_arch = "wasm32")'.dependencies]
wasm-bindgen = "0.2.76" # remember to change version in wiki as well
```

**Reply by sotrh** [reply]
Looks like "wgpu-hal" might be the culprit. It's using "0.2" for wasm-bindgen
```toml
[target.'cfg(target_arch = "wasm32")'.dependencies]
wasm-bindgen = { version = "0.2" }
```

**Comment by sotrh** [comment, marked as answer]
So the problem was that cargo was pulling the latest release of wasm-bindgen. The fix was to use `wasm-bindgen = "=0.2.76"` in the wgpu Cargo.toml to force cargo to use 0.2.76.

Changing wgpu-hal to specify 0.2.76 might fix it as well. I haven't checked.
